annotation/tests/test_tasks_services.py

```diff
@@ -585,3 +636,1697 @@ def test_filter_tasks_db_no_files_or_jobs(
+    ), patch(
+        "annotation.tasks.services.io.BytesIO"
+    ), patch(
+        "annotation.tasks.services.io.TextIOWrapper",
+    ):
+
+        filename, _ = create_export_csv(mock_db, schema, "tenant", "token")
+
+        assert filename == expected_filename
+
+
+def test_create_export_csv_no_annotation_stats(mock_db):
+    schema = MagicMock(
+        user_ids=[999],
+        date_from=datetime(2024, 1, 1),
+        date_to=datetime(2024, 1, 6),
+    )
+    filter_mock = (
+        mock_db.query.return_value.filter.return_value.filter.return_value
+    )
+    filter_mock.all.return_value = []
+
+    with patch(
+        "annotation.tasks.services.get_user_names_by_user_ids", return_value={}
+    ), patch(
+        "annotation.tasks.services.get_file_names_by_file_ids", return_value={}
+    ), patch(
+        "annotation.tasks.services.io.BytesIO"
+    ), patch(
+        "annotation.tasks.services.io.TextIOWrapper"
+    ):
+
+        with pytest.raises(HTTPException) as exc_info:
+            create_export_csv(mock_db, schema, "tenant", "token")
+
+        assert exc_info.value.status_code == 406
+        assert exc_info.value.detail == "Export data not found."
+
+
+@pytest.fixture
+def mock_task():
+    task = MagicMock(ManualAnnotationTask)
+    task.id = 1
+    task.user_id = 2
+    task.job_id = 10
+    task.file_id = 3
+    task.pages = {1, 2, 3}
+    task.is_validation = False
+    yield task
+
+
+@pytest.fixture
+def mock_tenant_data():
+    mock_data = MagicMock(TenantData)
+    mock_data.token = "mock_token"
+    yield mock_data
+
+
+def test_evaluate_agreement_score(
+    mock_session: Mock, mock_task: Mock, mock_tenant_data: Mock
+):
+    with patch(
+        "annotation.tasks.services.get_file_path_and_bucket",
+        return_value=(
+            "mock_s3_file_path",
+            "mock_s3_file_bucket",
+        ),
+    ) as mock_get_file_path_and_bucket, patch(
+        "annotation.tasks.services.get_agreement_score"
+    ) as mock_get_agreement_score, patch(
+        "annotation.tasks.services.compare_agreement_scores",
+        return_value=MagicMock(AgreementScoreComparingResult),
+    ) as mock_compare_agreement_scores:
+
+        mock_agreement_score_response = [
+            MagicMock(AgreementScoreServiceResponse)
+        ]
+        mock_get_agreement_score.return_value = mock_agreement_score_response
+        mock_session.query().all.return_value = [mock_task]
+        result = evaluate_agreement_score(
+            db=mock_session,
+            task=mock_task,
+            tenant="mock_tenant",
+            token=mock_tenant_data,
+        )
+
+        mock_get_file_path_and_bucket.assert_called_once_with(
+            mock_task.file_id, "mock_tenant", mock_tenant_data.token
+        )
+
+        mock_compare_agreement_scores.assert_called_once_with(
+            mock_agreement_score_response, AGREEMENT_SCORE_MIN_MATCH
+        )
+
+        assert isinstance(result, AgreementScoreComparingResult)
```

Comment:
what about asserting the content?

Reply:
The compare_agreement_scores function returns a mock, and the evaluate_agreement_score function also returns a mock. You mentioned that I shouldn't compare mocks directly.